src/odemis/driver/orsay.py

```diff
@@ -2656,22 +2687,255 @@ def __init__(self, name, role, parent, **kwargs):
+                               self.parent.datamodel.IonColumnMCS,
+                               self.parent.datamodel.Sed,
+                               self.parent.datamodel.HybridPlatform,
+                               self.parent.datamodel.Miss,)
+        # In case the preset mask already exist it is automatically overwritten.
+        self.parent.preset_manager.CreatePresetMask("Odemis-preset-mask", *relevant_components)
+        self.completePresetMaskName = "Odemis-preset-mask"  # Preset defined to load all the relevant components.
+
+        # TODO K.K. The specs of the high level aperture/preset describe the implementation of the probe current Use the
+        #  currently selected preset and combine with the faraday cup measurements and make an attribute/VA of the
+        #  probe current
+
     def terminate(self):
         """
         Called when Odemis is closed
         """
         pass
 
+    @roattribute
+    def hfw_nomag(self):
+        return self._hfw_nomag
+
+    def getAllPresetData(self):
+        preset_data = {}
+        for preset in self.parent.preset_manager.GetAllPresets().iter("Preset"):
+            if "_" not in preset.get("name"):
+                logging.warning(f"The preset {preset.get('name')} could not not be converted for use in Odemis. It "
+                                f"missed the necessary formatting.")
+                continue
+            full_preset = self.parent.preset_manager.GetPreset(preset.get("name"))
+            current = preset.get("name")[:preset.get("name").find("_")]
+
+            try:
+                preset_data.update({current:
+                                            {"name": preset.get("name"),
+                                             "aperture_number": self._getApertureNmbrFromPreset(full_preset),
+                                             "condenser_voltage": self._getCondenserVoltageFromPreset(full_preset)}})
+            except LookupError as error_message:
+                logging.warning(f"The preset {preset.get('name')} missed a necessary setting and is therefore not "
+                                f"imported. The following message was returned when retrieving the preset settings:\n",
+                                error_message)
+        return preset_data
+
+    def getPresetSetting(self, preset, sub_comp, setting, tag="Target"):
+        """
+        Get a setting from a preset XML ElementTree
+
+        :param preset (xml.etree.ElementTree.Element): Full preset XML ElementTree including sub elements with all sub
+        components
+        :param sub_comp (str): Name of the sub component
+        :param setting (str): Name of the parameter with the setting
+        :param tag (str): tag specifying the value to obtain (Most likely 'Target' or for example 'Max', 'Min',
+        'AtTarget', 'Tolerance')
+        :return (None/Str): Value found for the setting and None if no value is found.
+        """
+        for c in preset.getchildren():
```

Comment:
`.getchildren()` is deprecated (since Python 3.2). Apparently, the new way is to just write `for c in preset`.